Lecture4/src/test/java/CalcUnitTests.java

```diff
@@ -0,0 +1,142 @@
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class CalcUnitTests {
+	private double op1;
+	private double op2;
+	private Operator operator;
+
+	private Calculator newCalc() {
+		return new Calculator(op1, operator, op2);
+	}
+
+	private double getOperand(double operand) {
+		return operand;
+	}
+
+	private Operator getOperator(Operator operator) {
+		return operator;
+	}
+
+	@BeforeClass
+	public void oneTimeSetUp() {
+		newCalc();
+		System.out.print("Before Class: newCalc();");
+	}
+
+	@Test(dataProvider = "testAddDataProvider")
+	public void add_twoNumbers_returnsSum(double operand1, double operand2, double total) {
```

Comment:
not java convention naming 
`addTwoNumbersReturnsSum ` or better `checkSum`

Reply:
fixed